plugins/chat/assets/javascripts/discourse/components/full-page-chat.hbs

```diff
@@ -1,3 +1,3 @@
-{{#if @channel.id}}
-  <ChatChannel @channel={{@channel}} @targetMessageId={{@targetMessageId}} />
-{{/if}}
\ No newline at end of file
+{{#each (array @channel) as |channel|}}
+  <ChatChannel @channel={{channel}} @targetMessageId={{@targetMessageId}} />
+{{/each}}
```

Comment:
This changes ensures ember won't try to optimize the performance of components by reusing the same component without going through the whole did-insert/will-destroy lifecyle, which was forcing us to handle "did-update" channel and complicating logic a lot. Technically this is not directly related to this PR, but the changes made in this PR were making some of the specs more brittle because of this.